internal/querynodev2/delegator/segment_pruner.go

```diff
@@ -15,12 +15,12 @@ import (
 	"github.com/milvus-io/milvus/internal/storage"
 	"github.com/milvus-io/milvus/internal/util/clustering"
 	"github.com/milvus-io/milvus/internal/util/exprutil"
-	"github.com/milvus-io/milvus/internal/util/typeutil"
 	"github.com/milvus-io/milvus/pkg/common"
 	"github.com/milvus-io/milvus/pkg/log"
 	"github.com/milvus-io/milvus/pkg/util/distance"
 	"github.com/milvus-io/milvus/pkg/util/funcutil"
 	"github.com/milvus-io/milvus/pkg/util/merr"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
 )
 
 const defaultFilterRatio float64 = 0.5
```

Comment:
filter ratio might not be good enough, we need a lookup table here.
If segment <= 16 , do not filtering
[17, 32],  segment * 0.5
[33, 256],  16 + (segment - 32) * 0.2
[256, 1024], 60 +  (segment - 256) * 0.05
[1024, more], 98 + (segment - 1024) * 0.01

Reply:
and we should make sure this is tunable

Reply:
need to test recall on multiple dataset as well

Reply:
ideally we expect segment size to be 100+